### PR TITLE
Deployment support on native k8s clusters

### PR DESCRIPTION
**Pull request by fbladilo**
**DONE:**

* Enables auto detection of running cluster (k8s vs Openshift)
* Re-factored TLS support, forklift can now be deployed with or **_without_** TLS support: 
   - Each forklift sub-component can have TLS enabled or disabled via these variables (enabled by default) : inventory_tls_enabled , ui_tls_enabled and validation_tls_enabled
   - We continue to use OpenShift service serving cert facilities **_by default_** when TLS is enabled. But now we have the ability to customize certificate secrets via inventory_tls_secret_name , validation_tls_secret_name and ui_tls_secret_name, this allows k8s users to pick how they want to supply TLS keys/certs
* Removed unused forklift-controller service and made rest of exposed service ports more descriptive for both http/https protocols
* Added k8s deployment documentation

**NOTES:** 

The method used for auto detection of cluster is by checking the ClusterVersion resource which is unique to OpenShift, I picked this option over parsing all available APIs looking for route.openshift.io which was more taxing in performance and this is a task we will run on every reconcile. (it has to query all available APIs)

**TODO :** 
- Document/implement ingress setup

### CODE REVIEW

File: docs/k8s.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# Konveyor Forklift k8s Installation Instructions
+
+## Pre-requisites
+
+- **Kubernetes cluster or Minikube v1.17+**
+- **OLM**
```
Comment (fdupont-redhat) [inline docs/k8s.md:6]: Can you expand the OLM acronym? At least once at the top of the page, so people know what it stands for.

File: docs/k8s.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+### Installing OLM support
+
+We strongly suggest OLM support for Forklift deployments, in some production kubernetes clusters OLM might already be present, if not, follow official instructions in how to add it. Below is an example in how to add it to minikube.
+
+`minikube addons enable olm`
+
```
Comment (fdupont-redhat) [inline docs/k8s.md:13]: Can you also link to the OLM documentation for the deployment on vanilla k8s? And if it's really short, maybe add the steps directly here.

**Review by fdupont-redhat — CHANGES_REQUESTED** [review]
Overall looks good.
If running on vanilla k8s, let's skip the UI deployment completely. On k8s, the CLI/API rules and even k8s dashboard is messy, so it is natural to k8s users to not have a UI. When we have a better way to handle the authentication, we can reenable it.

I guess that the easiest way is to set the `feature_ui` variable to `false` when `k8s_cluster` is `true`.

**Comment by fbladilo** [comment]
> Overall looks good.
> If running on vanilla k8s, let's skip the UI deployment completely. On k8s, the CLI/API rules and even k8s dashboard is messy, so it is natural to k8s users to not have a UI. When we have a better way to handle the authentication, we can reenable it.
> 
> I guess that the easiest way is to set the `feature_ui` variable to `false` when `k8s_cluster` is `true`.

feature_ui is an extra variable passed to the operator , so manipulating (i.e via set_fact) will not have an effect because of ansible variable precedence rules (extras are highest priority). I can however expand the conditional to never deploy when k8s_cluster is true , similar to what I did for Oauth/CORS setup.

File: docs/k8s.md (hunk 1 above)
Comment (fbladilo) [reply docs/k8s.md]: Enhanced docs with more details , let me know...

File: docs/k8s.md (hunk 2 above)
Comment (fbladilo) [reply docs/k8s.md:20]: Done , see last commit

**Comment by fdupont-redhat** [comment]
I think that disabling UI when installing on k8s is the right thing to do, even if `feature_ui` is `true`. I let you decide what is the best option.

**Pull request merged**